modules/dbfs/filestorage.go

```diff
@@ -17,7 +17,7 @@ var filePathSeparator = strconv.QuoteRune(os.PathSeparator)[1:2]
 // FileWrite writes the file with the given bytes to a calculated path, and
 // returns that path so it can be put in MySQL
 func (di *DatabaseImpl) FileWrite(relpath string, filename string, projectID int64, raw []byte) (string, error) {
-	relFilePath, err := di.calculateFilePath(relpath, filename, projectID)
+	relFilePath, err := di.cleanPath(relpath, filename, projectID)
```

Comment:
maybe getPath(...)? cleanPath implies just cleaning up the filepath given, not constructing a new one.